### PR TITLE
Use makefat instead of lipo, and vendor the code of our build tools (golangci-lint, makefat, gomod2rpmdeps)

### PR DESCRIPTION
**Pull request by cfergeau**
Using makefat instead of lipo allows to cross-build universal binaries instead of having to do it on a macOS machine.
Vendoring golangci-lint/makefat/... allows to run 'make cross-lint'/... on systems without network access.

### CODE REVIEW

**Comment by anjannath** [comment]
on macOS (M1) `make cross-lint` is giving the following error, this is also happening in the `main` branch, but only noticed while looking at this PR
```
$ make cross-lint
GOOS=darwin /Users/anjan/work/crc/tools/bin/golangci-lint run
GOOS=linux /Users/anjan/work/crc/tools/bin/golangci-lint run
GOOS=windows /Users/anjan/work/crc/tools/bin/golangci-lint run
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package w32uiautomation: could not load export data: no export data for "github.com/openstandia/w32uiautomation"
ERRO Running error: 1 error occurred:
        * can't run linter goanalysis_metalinter: buildir: failed to load package w32uiautomation: could not load export data: no export data for "github.com/openstandia/w32uiautomation"

make: *** [cross-lint] Error 3
```

File: tools/go.mod [hunk 1]
```diff
@@ -0,0 +1,173 @@
+module github.com/code-ready/crc/tools
+
+go 1.18
```
Comment (anjannath) [inline tools/go.mod:3]: maybe we should keep the same go version as the one used for crc here? `1.17`

Comment (cfergeau) [reply tools/go.mod]: Yes, definitely, thanks for catching this! I've repushed with this changed, and will look at your m1 issue next.

**Comment by cfergeau** [comment]
> on macOS (M1) `make cross-lint` is giving the following error, this is also happening in the `main` branch, but only noticed while looking at this PR
> 
> ```
> $ make cross-lint
> GOOS=darwin /Users/anjan/work/crc/tools/bin/golangci-lint run
> GOOS=linux /Users/anjan/work/crc/tools/bin/golangci-lint run
> GOOS=windows /Users/anjan/work/crc/tools/bin/golangci-lint run
> WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package w32uiautomation: could not load export data: no export data for "github.com/openstandia/w32uiautomation"
> ERRO Running error: 1 error occurred:
>         * can't run linter goanalysis_metalinter: buildir: failed to load package w32uiautomation: could not load export data: no export data for "github.com/openstandia/w32uiautomation"
> 
> make: *** [cross-lint] Error 3
> ```

Fixed (or worked around depending on the perspective) by forcing GOARCH=amd64 for Windows golangci-lint runs.

**Review by anjannath — APPROVED** [review]

**Pull request merged**